src/rpdk/core/data_loaders.py

```diff
@@ -1,6 +1,7 @@
 import json
 import logging
 import os
+import re
```

Comment:
I like this type of testing.  Get rid of some of the java patterns would be helpful. My question is the python re the one we want to standardize on?  For instance `^[\d\w-_.+]*$` is technically valid but in python it needs to be `^[\d\w\-_.+]*$`.

Reply:
Ideally, resource schema patterns are valid and equivalent in both Python and Java

[JSON schema itself recommends sticking to a minimal subset of regular expression syntax](http://json-schema.org/understanding-json-schema/reference/regular_expressions.html). Let's encourage the same

Just emitting warnings for patterns not valid in Python seems like a good start in that direction

Reply:
> JSON schema itself recommends sticking to a minimal subset of regular expression syntax. Let's encourage the same

Agreed completely here.

Reply:
Golang has some interesting regex limitations even within that minimal subset:
https://github.com/hashicorp/terraform-provider-awscc/issues/88
https://github.com/golang/go/issues/7252